entrypoint/exec/go_init.go

```diff
@@ -0,0 +1,124 @@
+/*
+
+Shameless forked and editted version of `go-init`
+https://github.com/pablo-ruth/go-init/tree/30e47fe6f58c0e7fb5dfd12525b583639c7bda92
+
+MIT License
+
+Copyright (c) 2017 Pablo RUTH
+Copyright (c) 2020 Red Hat, Inc
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+*/
+
+package exec
+
+import (
+	"context"
+	"os"
+	"os/exec"
+	"os/signal"
+	"sync"
+	"syscall"
+	"time"
+)
+
+func RemoveZombies(ctx context.Context, wg *sync.WaitGroup) {
```

Comment:
I'd personally be more in favor of adding an assertion that we're not pid 1 (i.e. require that the caller has used `--init` or `dumb-init` in Kubernetes) and documenting how to do that rather than also carrying our own init.  But not a blocker.